### ISSUE TITLE
How should a server inform NVDA of changes in selection?

### ISSUE THREAD

**Issue by da40** [nvaccess/nvda#8219]
Apologies if this is the wrong forum to ask this question.

I wonder if anyone can help me with an issue I'm having adapting a text editor for screen reading. I've succeeded in getting NVDA to read back the line of text under the cursor, and the character after the caret, but cannot get it to handle selected text.  When text is selected, this call is made:

NotifyWinEvent(EVENT_OBJECT_SELECTION, hWnd, OBJID_CLIENT, CHILDID_SELF);

(The event is changed to EVENT_OBJECT_SELECTIONADD/REMOVE, as appropriate.)

The window has a handler with this signature, but it never gets called:

HRESULT get_accSelection(VARIANT *pvarChildren)

In case it's significant, get_accRole returns ROLE_SYSTEM_TEXT, get_accChildCount returns 0, and get_accState returns STATE_SYSTEM_SELECTED or STATE_SYSTEM_EXTSELECTABLE, as appropriate.

How should selection be handled?

Thanks, Keith

**Comment by Adriani90** [comment]
@da40 I suggest to write nv Access at info@nvaccess.org and ask for assistance in this regard in case this is still something that needs input.

I am closing this one as abandoned due to missing reactions from anyone.
